actors/power/src/lib.rs

```diff
@@ -617,49 +620,15 @@ impl Actor {
 }
 
 impl ActorCode for Actor {
-    fn invoke_method<RT>(
-        rt: &mut RT,
-        method: MethodNum,
-        params: &RawBytes,
-    ) -> Result<RawBytes, ActorError>
-    where
-        RT: Runtime,
-    {
-        match FromPrimitive::from_u64(method) {
-            Some(Method::Constructor) => {
-                Self::constructor(rt)?;
-                Ok(RawBytes::default())
-            }
-            Some(Method::CreateMiner) => {
-                let res = Self::create_miner(rt, cbor::deserialize_params(params)?)?;
-                Ok(RawBytes::serialize(res)?)
-            }
-            Some(Method::UpdateClaimedPower) => {
-                Self::update_claimed_power(rt, cbor::deserialize_params(params)?)?;
-                Ok(RawBytes::default())
-            }
-            Some(Method::EnrollCronEvent) => {
-                Self::enroll_cron_event(rt, cbor::deserialize_params(params)?)?;
-                Ok(RawBytes::default())
-            }
-            Some(Method::OnEpochTickEnd) => {
-                Self::on_epoch_tick_end(rt)?;
-                Ok(RawBytes::default())
-            }
-            Some(Method::UpdatePledgeTotal) => {
-                let param: TokenAmount = cbor::deserialize_params(params)?;
-                Self::update_pledge_total(rt, param)?;
-                Ok(RawBytes::default())
-            }
-            Some(Method::SubmitPoRepForBulkVerify) => {
-                Self::submit_porep_for_bulk_verify(rt, cbor::deserialize_params(params)?)?;
-                Ok(RawBytes::default())
-            }
-            Some(Method::CurrentTotalPower) => {
-                let res = Self::current_total_power(rt)?;
-                Ok(RawBytes::serialize(res)?)
-            }
-            None => Err(actor_error!(unhandled_message; "Invalid method")),
-        }
+    type Methods = Method;
+    actor_dispatch! {
+        Constructor => constructor,
+            CreateMiner => create_miner,
```

Comment:
indent

Reply:
I'm surprised rustfmt didn't kill me for this.